napari/layers/_shapes_layer/view/controls.py

```diff
@@ -0,0 +1,176 @@
+        layout.addWidget(self.delete_button)
+        layout.addStretch(0)
+        self.setLayout(layout)
+        self.setMouseTracking(True)
+
+        self.panzoom_button.setChecked(True)
+
+    def mouseMoveEvent(self, event):
+        self.layer.status = str(self.layer.mode)
+
+    def set_mode(self, event):
+        mode = event.mode
+        if mode == Mode.SELECT:
+            self.select_button.setChecked(True)
+        elif mode == Mode.DIRECT:
+            self.direct_button.setChecked(True)
+        elif mode == Mode.PAN_ZOOM:
+            self.panzoom_button.setChecked(True)
+        elif mode == Mode.ADD_RECTANGLE:
+            self.rectangle_button.setChecked(True)
+        elif mode == Mode.ADD_ELLPISE:
+            self.ellipse_button.setChecked(True)
+        elif mode == Mode.ADD_LINE:
+            self.line_button.setChecked(True)
+        elif mode == Mode.ADD_PATH:
+            self.path_button.setChecked(True)
+        elif mode == Mode.ADD_POLYGON:
+            self.polygon_button.setChecked(True)
+        elif mode == Mode.VERTEX_INSERT:
+            self.vertex_insert_button.setChecked(True)
+        elif mode == Mode.VERTEX_REMOVE:
+            self.vertex_remove_button.setChecked(True)
+        else:
+            raise ValueError("Mode not recongnized")
+
+
+class QtModeButton(QRadioButton):
+    def __init__(self, layer, button_name, mode, tool_tip):
+        super().__init__()
+
+        self.mode = mode
+        self.layer = layer
+        self.setToolTip(tool_tip)
+        self.setChecked(False)
+        styleSheet = button_style(button_name)
+        self.setStyleSheet(styleSheet)
+        self.toggled.connect(lambda state=self: self._set_mode(state))
+        self.setFixedWidth(28)
+
+    def _set_mode(self, bool):
+        with self.layer.events.mode.blocker(self._set_mode):
+            if bool:
+                self.layer.mode = self.mode
+
+
+class QtDeleteButton(QPushButton):
+    def __init__(self, layer):
+        super().__init__()
+
+        self.layer = layer
+        self.setIcon(QIcon(':/icons/delete.png'))
+        self.setFixedWidth(28)
+        self.setFixedHeight(28)
+        self.setToolTip('Delete selected')
+        self.clicked.connect(self.layer.remove_selected)
+
+
+class QtMoveBackButton(QPushButton):
+    def __init__(self, layer):
+        super().__init__()
+
+        self.layer = layer
+        self.setIcon(QIcon(':/icons/move_back.png'))
+        self.setFixedWidth(28)
+        self.setFixedHeight(28)
+        self.setToolTip('Move to back')
+        self.clicked.connect(self.layer.move_to_back)
+
+
+class QtMoveFrontButton(QPushButton):
+    def __init__(self, layer):
+        super().__init__()
+
+        self.layer = layer
+        self.setIcon(QIcon(':/icons/move_front.png'))
+        self.setFixedWidth(28)
+        self.setFixedHeight(28)
+        self.setToolTip('Move to front')
+        self.clicked.connect(self.layer.move_to_front)
+
+
+def button_style(name):
```

Comment:
@kne42 do you have ideas on how I can remove this? Right not I have a single class and then use this function to set all the stylesheets for the different buttons. Should I instead just make each button its own class? Then I could set the stylesheets in the way that you do in the `.qss` file

Reply:
Depends on how dynamic you would like this to be. Using the main stylesheet inherently limits you to specify exactly what buttons you can customize. Programmatically setting it would let you add as many button styles as you want at any time.

If you only have a set number of button types, the ideal way to do it would be to use the trick I talked about:
```python
self.setProperty('mode', button_name)
```

```css
QtModeButton {
  background-color:lightGray;
  border-radius: 3px;
}

QtModeButton::indicator {
  subcontrol-position:
  center center;
  subcontrol-origin: content;
  width: 28px; height: 28px;
}

QtModeButton::indicator:checked {
  background-color:rgb(0, 153, 255);
  border-radius: 3px;
}

...

QtModeButton[mode="..."]::indicator:checked {
  image: url(...);
}

...
```

Reply:
thanks!! this was so helpful. Fixed :-)